src/Workspaces/Core/Portable/Log/AnalyzerNameForTelemetry.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Internal.Log
+{
+    internal static class AnalyzerNameForTelemetry
+    {
+        public static string ComputeSha256Hash(string name)
+        {
+            using var sha256 = SHA256.Create();
+            return Convert.ToBase64String(sha256.ComputeHash(Encoding.UTF8.GetBytes(name)));
```

Comment:
any way to compute this without so many allocs?  bytes->hash->b64?  or do we only have a few of these?

Reply:
(is there anything that can take ROS<char> or ROS<byte>?

Reply:
We won't do this that often, and we're already doing it for analyzers which I think probably would outnumber generators. That said, I'll see what we can do.

Reply:
that's ok with me.  i wsan't certain if this was going to be called in bulk.

Reply:
Yeah, so this is ugly: since this has to run on .NET Framework we don't have many options; if rather than this being the hash of the UTF8 string it was UTF16 this could potentially be a lot better because we could just directly walk the underlying buffer but I don't think we have good options.

I'll defer this one since it'd be moot if we can just switch to PiiValue.

Reply:
sounds fine.